de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/session/JoinSessionRequestHandler.java

```diff
@@ -0,0 +1,61 @@
+package de.fu_berlin.inf.dpp.server.session;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.jivesoftware.smack.PacketListener;
+import org.jivesoftware.smack.packet.Packet;
+
+import de.fu_berlin.inf.dpp.communication.extensions.JoinSessionRequestExtension;
+import de.fu_berlin.inf.dpp.net.IReceiver;
+import de.fu_berlin.inf.dpp.net.xmpp.JID;
+import de.fu_berlin.inf.dpp.session.ISarosSessionManager;
+import de.fu_berlin.inf.dpp.util.NamedThreadFactory;
+
+/**
+ * Listens for and handles JoinSessionRequests allowing clients to join server sessions.
+ */
+public final class JoinSessionRequestHandler {
+    private final ISarosSessionManager sessionManager;
+
+    /**
+     * Invitation Requests are handled very fast and are almost immediately dispatched
+     * into another executor in the {@link NegotiationHandler}. More then one thread is currently
+     * not necessary and would likely just introduce more overhead.
+     * We just want asynchronous execution and a queue.
+     */
+    private final ExecutorService executor = Executors.newSingleThreadExecutor(
```

Comment:
I think we have to redesign some parts. There is currently too much threading involved. It does not makes that every class has its own threadpool.

The problem here is that you do not restrict the queue to a size. I can simply start a (D)DoS attack by spamming the server with request packets.

Reply:
Well I guess there a lot of cases, where we could easily ddos saros by misusing the protocol. But sure, I will fix this one.